konan_sdk/endpoints/konan_endpoints.py

```diff
@@ -1,52 +1,120 @@
 
-        if deployment_uuid is None:
-            raise ValueError("A valid deployment_uuid must be specified")
-        super().__init__(api_url=api_url, user=user)
 
-        self.deployment_uuid = deployment_uuid
+class PredictionEndpoint(KonanBaseDeploymentEndpoint):
+    @property
+    def name(self) -> str:
+        return 'predict'
 
     @property
-    def headers(self):
-        return {
-            'Authorization': f"Bearer {self.user.access_token}",
-            'Content-Type': 'application/json',
-        }
+    def endpoint_path(self) -> str:
+        return super().endpoint_path + '/predict'
+
+    class RequestObject():
+        def __init__(self, input_data: Union[Dict, str]) -> None:
+            self.input_data = input_data
+
+    class ResponseObject():
+        def __init__(self, prediction_uuid: str, output: Any) -> None:
+            self.prediction_uuid = prediction_uuid
+            self.output = output
+
+    def prepare_request(self, request_object: RequestObject) -> KonanEndpointRequest:
+        return KonanEndpointRequest(data=request_object.input_data)
 
+    def process_response(self, endpoint_response: KonanEndpointResponse) -> ResponseObject:
+        return PredictionEndpoint.ResponseObject(prediction_uuid=endpoint_response.json['prediction_uuid'], output=endpoint_response.json['output'])
+
+
+class EvaluateEndpoint(KonanBaseDeploymentEndpoint):
     @property
-    def endpoint_path(self):
-        return f"/deployments/{self.deployment_uuid}/predict/"
+    def name(self) -> str:
+        return 'evaluate'
 
-    def process_response(self):
-        prediction_uuid = self.response.pop('prediction_uuid')
-        self.response = prediction_uuid, self.response
+    @property
+    def endpoint_path(self) -> str:
+        return super().endpoint_path + '/evaluate'
+
+    class RequestObject():
+        def __init__(self, start_time: datetime.datetime, end_time: datetime.datetime) -> None:
+            self.start_time = start_time
+            self.end_time = end_time
+
+    class ResponseObject():
+        def __init___(self, metrics: List[KonanBaseMetric], error: str = None) -> None:
+            self.metrics = metrics
+            self.error = error
+
+    def prepare_request(self, request_object: RequestObject) -> KonanEndpointRequest:
+        return KonanEndpointRequest(data={
+            'start_time': request_object.start_time.isoformat(),
+            'end_time': request_object.end_time.isoformat()
+        })
+
+    def process_response(self, endpoint_response: KonanEndpointResponse) -> ResponseObject:
+        predfined_metrics_dict = endpoint_response.json['metrics'].get("predefined", dict())
+        custom_metrics_list = endpoint_response.json['metrics'].get("custom", list())
+
+        metrics: List[KonanBaseMetric] = [
+            PredefinedMetrics.get(metric_name, KonanCustomMetric)(predfined_metrics_dict[metric_name], name=metric_name)
+            for metric_name in predfined_metrics_dict.keys()
```

Comment:
By the way, really small minor point but you can just do `for metric_name, value in predefined_metrics_dict.items()` here

But should we actually handle it as such?As in if the API returns it as Predefined and SDK classifies it as custom if its not defined. I am not sure if we should do that or error out instead

Reply:
I implemented this as a fallback in case, in the future, the API adds a new `PredefinedMetric`, while the SDK not yet updated to explicitly support it.

Note that the API adding a PredefinedMetric is **NOT** a breaking change, while removing one is. That's why I elected to add a fallback for _extra_ predefined metrics.